src/metric.jl

```diff
@@ -200,36 +229,37 @@ end
 """
     MSE
 
-Mean Squared Error. TODO: add support for multi-dimensional outputs.
+Mean Squared Error.
 
-Calculates the mean squared error regression loss in one dimension.
+Calculates the mean squared error regression loss.
+Requires that label and prediction have the same shape.
 """
 
 type MSE <: AbstractEvalMetric
-  mse_sum  :: Float64
+  mse_sum  :: Vector{NDArray}
   n_sample :: Int
 
-  MSE() = new(0.0, 0)
+  MSE() = new(Vector{NDArray}(), 0)
 end
 
-function _update_single_output(metric :: MSE, label :: NDArray, pred :: NDArray)
-  label = copy(label)
-  pred  = copy(pred)
-
-  n_sample = size(pred)[end]
-  metric.n_sample += n_sample
+hasNDArraySupport(::MSE) = Val{true}()
 
-  for i = 1:n_sample
-    metric.mse_sum += (label[i] - pred[i])^2
-  end
+function _update_single_output(metric :: MSE, label :: NDArray, pred :: NDArray)
+  @assert size(label) == size(pred)
+  metric.n_sample += length(label)
+  mse_sum = mx.sum(mx._PowerScalar(label - pred,scalar=2))
+  push!(metric.mse_sum, mse_sum)
+  return nothing
 end
 
 function get(metric :: MSE)
-  return [(:MSE, metric.mse_sum / metric.n_sample)]
+  # Delay copy until last possible moment
+  mse_sum = mapreduce(nda->copy(nda)[1], +, 0.0, metric.mse_sum)
```

Comment:
My understanding is that each `NDArray` in `mse_sum` at this point are actually scalars? *Maybe* using `asscalar` will be more efficient than the generic `copy` function. Otherwise, LGTM.